### ISSUE TITLE
RECOMMEND/VOTE Papers for the 2nd session

### ISSUE THREAD

**Issue by sxjscience** [ML-HK/paper-discussion-group#2]
We can recommend some papers for further discussion under this issue. Include a link to the paper + the conference name and other related information (like the abstract, some basic descriptions, links to samples code or online demonstrations).

Please only include one topic per comment. For example, if you propose to discuss "paper X" which is heavily based on "paper Y" and you believe both have to be read together (possibly over multiple weeks) just create one comment for that. If you propose two unrelated papers please create two comments.

The voting period ends up on 2016/09/25. We are free to recommend new papers.

Please vote using the "Thumbs up" emoji.

**Comment by sxjscience** [comment]
Continue reading the paper recommended in previous session.

[**WaveNet: A Generative Model for Raw Audio**](http://arxiv.org/abs/1609.03499)
(arXiv - submitted on 12 Sep 2016)
 _This paper introduces WaveNet, a deep neural network for generating raw audio waveforms. The model is fully probabilistic and autoregressive, with the predictive distribution for each audio sample conditioned on all previous ones; nonetheless we show that it can be efficiently trained on data with tens of thousands of samples per second of audio. When applied to text-to-speech, it yields state-of-the-art performance, with human listeners rating it as significantly more natural sounding than the best parametric and concatenative systems for both English and Mandarin. A single WaveNet can capture the characteristics of many different speakers with equal fidelity, and can switch between them by conditioning on the speaker identity. When trained to model music, we find that it generates novel and often highly realistic musical fragments. We also show that it can be employed as a discriminative model, returning promising results for phoneme recognition._ 

Also read
- [Conditional Image Generation with PixelCNN Decoders](http://arxiv.org/abs/1606.05328) (NIPS 2016)

**Comment by leezu** [comment]
During our last discussion this paper was mentioned.

[**On the difficulty of training recurrent neural networks**](http://jmlr.org/proceedings/papers/v28/pascanu13.pdf) (Razvan Pascanu, Tomas Mikolov, Yoshua Bengio)
(ICML2013 )
_There are two widely known issues with properly training recurrent neural networks, the vanishing and the exploding gradient problems detailed in Bengio et al. (1994). In this paper we attempt to improve the understanding of the underlying issues by exploring these problems from an analytical, a geometric and a dynamical systems perspective. Our analysis is used to justify a simple yet effective solution. We propose a gradient norm clipping strategy to deal with exploding gradients and a soft constraint for the vanishing gradients problem. We validate empirically our hypothesis and proposed solutions in the experimental section._

**Comment by leezu** [comment]
[**Highway Networks**](https://arxiv.org/abs/1505.00387) (Rupesh Kumar Srivastava, Klaus Greff, Jürgen Schmidhuber)
 (arXiv - submitted on 3 May 2015)
_There is plenty of theoretical and empirical evidence that depth of neural networks is a crucial ingredient for their success. However, network training becomes more difficult with increasing depth and training of very deep networks remains an open problem. In this extended abstract, we introduce a new architecture designed to ease gradient-based training of very deep networks. We refer to networks with this architecture as highway networks, since they allow unimpeded information flow across several layers on "information highways". The architecture is characterized by the use of gating units which learn to regulate the flow of information through a network. Highway networks with hundreds of layers can be trained directly using stochastic gradient descent and with a variety of activation functions, opening up the possibility of studying extremely deep and efficient architectures._

**Comment by leezu** [comment]
[**Training Very Deep Networks**](https://papers.nips.cc/paper/5850-training-very-deep-networks.pdf) (Rupesh Kumar Srivastava, Klaus Greff, Jürgen Schmidhuber)
(NIPS 2015)
_Theoretical and empirical evidence indicates that the depth of neural networks is crucial for their success. However, training becomes more difficult as depth increases, and training of very deep networks remains an open problem. Here we introduce a new architecture designed to overcome this. Our so-called highway networks allow unimpeded information flow across many layers on information highways. They are inspired by Long Short-Term Memory recurrent networks and use adaptive gating units to regulate the information flow. Even with hundreds of layers, highway networks can be trained directly through simple gradient descent. This enables the study of extremely deep and efficient architectures._

**Comment by sxjscience** [comment]
[**Dueling Network Architectures for Deep Reinforcement Learning**](http://jmlr.org/proceedings/papers/v48/wangf16.pdf) (Ziyu Wang, Tom Schaul, Matteo Hessel, Hado van Hasselt, Marc Lanctot, Nando de Freitas)
(ICML 2016)
_In recent years there have been many successes of using deep representations in reinforcement learning. Still, many of these applications use conventional architectures, such as convolutional networks, LSTMs, or auto-encoders. In this paper, we present a new neural network architecture for model-free reinforcement learning. Our dueling network represents two separate estimators: one for the state value function and one for the state-dependent action advantage function. The main benefit of this factoring is to generalize learning across actions without imposing any change to the underlying reinforcement learning algorithm. Our results show that this architecture leads to better policy evaluation in the presence of many similar-valued actions. Moreover, the dueling architecture enables our RL agent to outperform the state-of-the-art on the Atari 2600 domain._

**Comment by yzhangee** [comment]
[Variational Autoencoder for Deep Learning of Images, Labels and Captions](http://people.ee.duke.edu/~lcarin/Yunchen_nips_2016.pdf) (Yunchen Pu, Zhe Gan, Ricardo Henao, Xin Yuan, Chunyuan Li, Andrew Stevens and Lawrence Carin)
(NIPS 2016)
_A novel variational autoencoder is developed to model images, as well as associated
labels or captions. The Deep Generative Deconvolutional Network (DGDN) is used
as a decoder of the latent image features, and a deep Convolutional Neural Network
(CNN) is used as an image encoder; the CNN is used to approximate a distribution
for the latent DGDN features/code. The latent code is also linked to generative
models for labels (Bayesian support vector machine) or captions (recurrent neural
network). When predicting a label/caption for a new image at test, averaging is
performed across the distribution of latent codes; this is computationally efficient as
a consequence of the learned CNN-based encoder. Since the framework is capable
of modeling the image in the presence/absence of associated labels/captions, a
new semi-supervised setting is manifested for CNN learning with images; the
framework even allows unsupervised CNN learning, based on images alone._

**Comment by yyuanad** [comment]
[InfoGAN: Interpretable Representation Learning by Information Maximizing Generative Adversarial Nets](https://arxiv.org/pdf/1606.03657v1.pdf) (Xi Chen, Yan Duan, Rein Houthooft, John Schulman, Ilya Sutskever, Pieter Abbeel)
(NIPS2016)
This paper describes InfoGAN, an information-theoretic extension to the Generative
Adversarial Network that is able to learn disentangled representations in a
completely unsupervised manner. InfoGAN is a generative adversarial network
that also maximizes the mutual information between a small subset of the latent
variables and the observation. We derive a lower bound of the mutual information
objective that can be optimized efficiently. Specifically, InfoGAN successfully
disentangles writing styles from digit shapes on the MNIST dataset, pose from
lighting of 3D rendered images, and background digits from the central digit on
the SVHN dataset. It also discovers visual concepts that include hair styles, presence/absence
of eyeglasses, and emotions on the CelebA face dataset. Experiments
show that InfoGAN learns interpretable representations that are competitive with
representations learned by existing supervised methods.

**Comment by yyuanad** [comment]
[Harnessing Deep Neural Networks with Logic Rules](https://www.cs.cmu.edu/~zhitingh/data/acl16harnessing.pdf)(Zhiting Hu, Xuezhe Ma, Zhengzhong Liu, Eduard Hovy, Eric P. Xing)
(ACL2016)(one of the 10 outstanding papers)
Combining deep neural networks with structured logic rules is desirable to harness flexibility and reduce uninterpretability of the neural models. We propose a general framework capable of enhancing various types of neural networks (e.g., CNNs and RNNs) with declarative first-order logic rules. Specifically, we develop an iterative distillation method that transfers the structured
information of logic rules into the weights of neural networks. We deploy the framework on a CNN for sentiment analysis, and an RNN for named entity recognition. With a few highly intuitive rules, we obtain substantial improvements and achieve state-of-the-art or comparable results to previous best-performing systems.

**Comment by leezu** [comment]
Thank you for the votes and the proposals.

As we agreed after the last session it makes more sense to have one person that leads through the discussion and prepares a short presentation for the main points. As I proposed the paper I will prepare this for Friday.

I think we can copy the other proposed papers that got at least one vote to the issue for next week (?).
